common_core.c

```diff
@@ -131,4 +131,51 @@ int ta_find_transactions_by_tag(const iota_client_service_t* const service,
   return ret;
 }
 
-int ta_get_txn_msg(const flex_hash_array_t* const txn) { return 0; }
+int ta_get_txn_msg(const iota_client_service_t* const service,
+                   const ta_get_transaction_msg_req_t* req,
+                   ta_get_transaction_msg_res_t* res) {
+  if (req == NULL || res == NULL) {
+    return -1;
+  }
+
+  retcode_t ret = RC_OK;
+  iota_transaction_t* tx;
+  flex_trit_t* tx_trits;
+  size_t num_hash = hash243_queue_count(req->hashes);
```

Comment:
I defer to the input of @ender503 to clarify the assumption for `num_hash`.

Reply:
This API only retrieve the message from one transaction, and we should not use it with the claim.
Since the claim may contain the `message signature`, it needs multiple transactions to store it.
It would be great to use the bundle to save the claim message by another API `ta_get_bundle_msg`.